iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/repository/HouseTableRepositoryImpl.java

```diff
@@ -107,23 +110,20 @@ public HouseTable save(HouseTable entity) {
     CreateUpdateEntityRequestBodyUserTable requestBody =
         new CreateUpdateEntityRequestBodyUserTable().entity(houseTableMapper.toUserTable(entity));
 
-    return getHtsRetryTemplate(Arrays.asList(IllegalStateException.class))
-        .execute(
-            context ->
-                apiInstance
-                    .putUserTable(requestBody)
-                    .map(EntityResponseBodyUserTable::getEntity)
-                    .map(houseTableMapper::toHouseTable)
-                    .onErrorResume(this::handleHtsHttpError)
-                    .block(Duration.ofSeconds(REQUEST_TIMEOUT_SECONDS)));
+    return apiInstance
+        .putUserTable(requestBody)
+        .map(EntityResponseBodyUserTable::getEntity)
+        .map(houseTableMapper::toHouseTable)
+        .onErrorResume(this::handleHtsHttpError)
+        .block(Duration.ofSeconds(WRITE_REQUEST_TIMEOUT_SECONDS));
   }
 
   @Override
   public Optional<HouseTable> findById(HouseTablePrimaryKey houseTablePrimaryKey) {
 
     return getHtsRetryTemplate(
             Arrays.asList(
-                HouseTableRepositoryStateUnkownException.class, IllegalStateException.class))
+                HouseTableRepositoryStateUnknownException.class, IllegalStateException.class))
```

Comment:
What was the case leading to IllegalStateException?

Reply:
https://github.com/reactor/reactor-core/issues/3519

Timeout in Mono.block() call leads to IllegalStateException